devtools/mog/generate.go

```diff
@@ -172,20 +164,39 @@ func sourceFieldMap(fields []fieldConfig) map[string]fieldConfig {
 }
 
 type generated struct {
-	To   *ast.FuncDecl
-	From *ast.FuncDecl
+	To                     *ast.FuncDecl
+	PtrTo, ToPtr, PtrToPtr *ast.FuncDecl
+
+	From                         *ast.FuncDecl
+	PtrFrom, FromPtr, PtrFromPtr *ast.FuncDecl
 
 	// TODO: RoundTripTest *ast.FuncDecl
 }
 
+func (g *generated) Decls() []ast.Decl {
+	return []ast.Decl{
```

Comment:
This will likely generate a bunch of dead code, unless the pointerness is inlined into the caller which we could improve upon later.

Reply:
Mentioned in slack, but I am wondering if we should change the default function definitions to something like the below, so that we can use a single set of functions for all of these:

```
func SourceFromTarget(t *Target, s *Source) {}
func SourceToTarget(s *Source, *Target) {}
```

Reply:
Pushing a followup commit that does just that^^